src/system/applications/combat/combat-tracker.ts

```diff
@@ -0,0 +1,151 @@
+        );
+    }
+
+    //toggles combatant turn speed on clicking the "fast/slow" text on the combat tracker window
+    protected _onClickToggleTurnSpeed(event: Event) {
+        event.preventDefault();
+        event.stopPropagation();
+        const btn = event.currentTarget as HTMLElement;
+        const li = btn.closest<HTMLElement>('.combatant')!;
+        const combatant: CosmereCombatant = this.viewed!.getEmbeddedDocument(
+            'Combatant',
+            li.dataset.combatantId!,
+            {},
+        ) as CosmereCombatant;
+        void combatant.toggleTurnSpeed();
+    }
+
+    //activates the combatant when clicking the activation icon
+    protected _onActivateCombatant(event: Event) {
+        event.preventDefault();
+        event.stopPropagation();
+        const btn = event.currentTarget as HTMLElement;
+        const li = btn.closest<HTMLElement>('.combatant')!;
+        const combatant: CosmereCombatant = this.viewed!.getEmbeddedDocument(
+            'Combatant',
+            li.dataset.combatantId!,
+            {},
+        ) as CosmereCombatant;
+        void combatant.setFlag('cosmere-rpg', 'activated', true);
+    }
+
+    //toggles combatant turn speed on clicking the "fast/slow" option in the turn tracker context menu
+    protected _onContextToggleTurnSpeed(li: JQuery<HTMLElement>) {
+        const combatant: CosmereCombatant = this.viewed!.getEmbeddedDocument(
+            'Combatant',
+            li.data('combatant-id') as string,
+            {},
+        ) as CosmereCombatant;
+        combatant.toggleTurnSpeed();
+    }
+
+    //resets combatants activation status to hasn't activated
+    protected _onContextResetActivation(li: JQuery<HTMLElement>) {
+        const combatant: CosmereCombatant = this.viewed!.getEmbeddedDocument(
+            'Combatant',
+            li.data('combatant-id') as string,
+            {},
+        ) as CosmereCombatant;
+        void combatant.setFlag('cosmere-rpg', 'activated', false);
+    }
+
+    _getEntryContextOptions(): ContextMenuEntry[] {
+        const menu: ContextMenuEntry[] = [
+            {
+                name: 'COSMERE.Combat.ToggleTurn',
+                icon: '',
+                callback: this._onContextToggleTurnSpeed.bind(this),
+            },
+            {
+                name: 'COSMERE.Combat.ResetActivation',
+                icon: '<i class="fas fa-undo"></i>',
+                callback: this._onContextResetActivation.bind(this),
+            },
+        ];
+        menu.push(
+            ...super
+                ._getEntryContextOptions()
+                .filter(
+                    (i) =>
+                        i.name !== 'COMBAT.CombatantReroll' &&
+                        i.name !== 'COMBAT.CombatantClear',
+                ),
+        );
+        return menu;
+    }
+}
+
+interface CosmereTurn {
+    id: string;
+    css: string;
+    pending: number;
+    finished: number;
+    type?: string;
+    turnSpeed?: string;
```

Comment:
Turn speed should be limited to only be `'fast'` or `'slow'`. You can do this by either making it a literal type or by using an enum.

**Literal**
```ts
type TurnSpeed = 'fast' | 'slow';
```

**Enum**
```ts
const enum TurnSpeed {
    Fast = 'fast',
    Slow = 'slow'
}
```